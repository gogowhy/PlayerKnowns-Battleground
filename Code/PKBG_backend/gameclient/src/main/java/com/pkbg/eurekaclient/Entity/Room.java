package com.pkbg.eurekaclient.Entity;

import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "room")
public class Room {

    public Integer roomnumber;

    public String hostname;

    public Integer playernumber;

    public Integer gamestatus;

    public Integer roompassword;

    public Integer getRoomnumber() {
        return roomnumber;
    }

    public void setRoomnumber(Integer roomnumber) {
        this.roomnumber = roomnumber;
    }

    public String getHostname() {
        return hostname;
    }

    public void setHostname(String hostname) {
        this.hostname = hostname;
    }

    public Integer getPlayernumber() {
        return playernumber;
    }

    public void setPlayernumber(Integer playernumber) {
        this.playernumber = playernumber;
    }

    public Integer getGamestatus() {
        return gamestatus;
    }

    public void setGamestatus(Integer gamestatus) {
        this.gamestatus = gamestatus;
    }

    public Integer getRoompassword() {
        return roompassword;
    }

    public void setRoompassword(Integer roompassword) {
        this.roompassword = roompassword;
    }
}
