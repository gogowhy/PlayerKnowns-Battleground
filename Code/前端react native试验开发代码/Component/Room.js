import React, { Component } from 'react';
import {
    Text,
    View,
    StyleSheet,
    TouchableOpacity,
    Alert,
    SectionList,
    FlatList,
    ImageBackground,
    Dimensions,
    Platform,
    BackHandler
} from 'react-native';
import base from '../src/style/base';
import header from '../src/style/header';
import Ionicons from "react-native-vector-icons/Ionicons";
import AntDesign from "react-native-vector-icons/AntDesign";
import Entypo from "react-native-vector-icons/Entypo";

const {height,width} = Dimensions.get('window')

/**
 * 该常量初步定义了一个players应该具有、并在render中有所体现的属性
 * 
 * 属性 ： team \  name  \  isReady
 * 属性 ： 组别  、 用户名 、 是否准备
 *
const players = [
    {team : 'A' , name : 'wang haoyu' , isReady : false},
    {team : 'B' , name : 'zhou yifan' , isReady : true},
    {team : 'A' , name : 'qi peng' , isReady : true},
    {team : 'A' , name : 'xie yihan' , isReady : true}
]
*/
/** 定义了同后端传递和接收指令的 Code 用来处理不同种类的消息 */
const START_GAME = 0 , READY = 1 , UNREADY = 2 , EXIT_BY_HOST = 3 , EXIT_BY_USER = 4 , CHANGE_TEAM_TO_A = 5 , CHANGE_TEAM_TO_B = 6 , NEW_PLAYER_JOIN = 7 , INITIALIZE = 8;

const DISMISS = 90 ,KICK = 91;
/** 定义了队伍常量 */
const TEAM_A = 1 , TEAM_B = 2;


export default class Room extends Component {

    constructor(props){
        super(props);
        this.state = {
            roomID : this.props.navigation.state.params.roomID , //房间ID
            password : this.props.navigation.state.params.password , //房间密码
            host : this.props.navigation.state.params.host , //是否为房主
            isReady : this.props.navigation.state.params.host , //是否准备
            username : this.props.navigation.state.params.username , //用户名
            players : [], //房间所有玩家
            socketState : WebSocket.CLOSED, //socket的连接状态
            team : TEAM_A, //玩家所处队伍
        }

        
        this.ws = new WebSocket("ws://49.234.27.75:2002/myHandler/ID="+this.state.username+"ROOMNUMBER="+this.state.roomID);
        
        this.ConnectWebSocket = this.ConnectWebSocket.bind(this);
        this.gobackMainPage = this.gobackMainPage.bind(this);
        this.startGame = this.startGame.bind(this);
        this.ready = this.ready.bind(this);
        this.unready = this.unready.bind(this);
        this.exitRoom = this.exitRoom.bind(this);
        this.enterGame = this.enterGame.bind(this);
        this.changeTeam = this.changeTeam.bind(this);
    }
    
    componentDidMount(){
        this.ConnectWebSocket();
    }
    /** 对物理返回键设置监听 */
    componentWillMount() {
        if (Platform.OS === 'android') {
            BackHandler.addEventListener("back", this.gobackMainPage);
        }else {

        }
    }

    componentWillUnmount() {
        if (Platform.OS === 'android') {
            BackHandler.removeEventListener("back", this.gobackMainPage);
        }else {
            this.propstManger.addLengeData(this.props.navigator.getCurrentRoutes().length);
        }
    }
    
    /**
     * 打开了同后端的 WebSocket 的连接
     * 定义了同后端连接的方法
     * 
     *
     * 
     */
    ConnectWebSocket(){
        
        /** 定义了同后端连接的 url 并建立连接 */
        
        this.ws.onopen = () => {
            // connection opened
            
            if(this.state.host)  //如果是房主，创建新房间的时候不会收到后端的 INITIALIZE 消息，需要自行将自己的信息 push 进 players列表里
            {
                var newplayers = [];
                newplayers.push({ username : this.state.username , playerstatus : true , playerteam : TEAM_A });
                
                this.setState(
                    {
                        players : newplayers ,
                        socketState : WebSocket.OPEN
                    }
                )
                
            }
            else{
                this.setState(
                    {
                        socketState : WebSocket.OPEN
                    }
                )
            }
        };
        
        this.ws.onmessage = (e) => {
        // a message was received
        if(this.ws.readyState !== WebSocket.OPEN) return;
        alert(e.data);
        let res1 = JSON.parse(e.data);
        //alert(str);
        let res = res1.pop();
        switch(res.code) {
            // 进入游戏
            case START_GAME : {
                alert("正在进入游戏！");
                this.enterGame();break;
            }
            // 设置某玩家准备
            case READY : {
                const username = res.username;
                var newplayers = this.state.players;
                newplayers.forEach( (player) => {
                    if(player.username == username){
                        player.playerstatus = 1 ;
                        return;
                    }
                }
                )

                this.setState(
                    { players : newplayers }
                )
                break;
            } 
            // 设置某玩家取消准备
            case UNREADY : {
                const username = res.username;
                var newplayers = this.state.players;
                newplayers.forEach( (player) => {
                    if(player.username == username){
                        player.playerstatus = 0 ;
                        return;
                    }
                }
                )

                this.setState(
                    { players : newplayers }
                )
                break;
            } 

            // 房主退出了房间
            case EXIT_BY_HOST : {
                const username = res.username;
                const newhostname = res.hostname;
                var newplayers = this.state.players;

                
                for(var i = 0 ;i < newplayers.length ; i++){
                    if(newplayers[i].username == username)
                    {
                        newplayers.splice(i,1);
                        break;
                    }
                }

                newplayers.forEach( (player) => {
                    if(player.username == newhostname){
                        player.playerstatus = 1 ;
                        return;
                    }
                }
                )

                if(newhostname == this.state.username){
                    this.setState({
                        host : true,
                        isReady : true
                    })
                }
                this.setState(
                    {
                        players : newplayers,
                    }
                )
                break;
            }
            // 非房主退出了房间
            case EXIT_BY_USER : {
                const username = res.username;
                var newplayers = this.state.players;

                for(var i = 0 ;i < newplayers.length ; i++){
                    if(newplayers[i].username == username)
                    {
                        newplayers.splice(i,1);
                        break;
                    }
                }
                this.setState(
                    {
                        players : newplayers,
                    }
                )
                break;
            }
            // 设置某玩家更换为A队
            case CHANGE_TEAM_TO_A : {
                const username = res.username;
                var newplayers = this.state.players;
                newplayers.forEach( (player) => {
                    if(player.username == username){
                        player.playerteam = TEAM_A ;
                        return;
                    }
                }
                )

                if(res.username == this.state.username){
                    this.setState(
                        {
                            team : TEAM_A
                        }
                    )
                }

                this.setState(
                    { players : newplayers }
                )
                break;
            }
            // 设置某玩家更换为B队
            case CHANGE_TEAM_TO_B : {
                const username = res.username;
                var newplayers = this.state.players;
                newplayers.forEach( (player) => {
                    if(player.username == username){
                        player.playerteam = TEAM_B ;
                        return;
                    }
                }
                )

                if(res.username == this.state.username){
                    this.setState(
                        {
                            team : TEAM_B
                        }
                    )
                }

                this.setState(
                    { players : newplayers }
                )
                break;
            }
            //  新玩家加入房间
            case NEW_PLAYER_JOIN : {
                
                let newplayer = {
                    username : res.username,
                    playerstatus : false,
                    playerteam : res.playerteam,
                }
                var newplayers = this.state.players;
                newplayers.push(newplayer);

                this.setState(
                    { players : newplayers }
                )
                break;
            }
            // 获取初始值
            case INITIALIZE : {

                var newplayers = res.players;
                newplayers.push({username : this.state.username , playerteam : res.playerteam , playerstatus : false }); //把自己的信息添加到列表里
                this.setState(
                    {
                        team : res.playerteam == 1 ? TEAM_A : TEAM_B,
                        players : newplayers
                    }
                )
                break;
            }
            // 房间被强制解散
            case DISMISS : {
                alert("该房间被强制解散！");
                this.gobackMainPage();
                break;
            }
            // 自己被踢出房间
            case KICK : {
                alert("你已被强制请出房间！");
                this.gobackMainPage();
                break;
            }
            // 报错
            case -1 : {
                if(res.message)
                    alert(res.message);
                else
                    alert("发生了一个意料之外的错误，请稍后再试。");
                break;
            }
        }
        };
        
        this.ws.onerror = (e) => {
        // an error occurred
        console.log(e.message);
        };
        
        this.ws.onclose = (e) => {
        // connection closed
        console.log(e.code, e.reason);
        };
    
    }

    /** 返回上一个界面 */
    gobackMainPage(){
        this.ws.close();
        this.setState({
            socketState : this.ws.readyState
        });
        const { goBack } = this.props.navigation ;
        goBack();
    }

    /** 进入游戏 */
    enterGame(){
        this.ws.close();
        this.setState({
            socketState : this.ws.readyState
        });
        const { navigate } = this.props.navigation ;
        navigate('Gaming',{username : this.state.username , team : this.state.team ,roomID : this.state.roomID });
    }

    /** 
     * 功能 ： 开始游戏
     * 触发 ： 房主点击“开始游戏”按钮
     * 
     * 检测是否还有玩家仍未准备，若无则向后端发送消息
     */
    startGame(){
        var flag = 1;
        this.state.players.forEach((player) => {
            if(!player.playerstatus) {alert("还有玩家未准备"); flag = 0; return; }
        }
        )
        /*
        const { navigate } = this.props.navigation ;
        navigate('Gaming',{username : this.state.username , team : this.state.team ,roomID : this.state.roomID });
        */
        if(flag)
        {   
            /*
            此处向后端发送数据，该房间将进入游戏
            */
            let data = {
                code : START_GAME , // START_GAME为数字常量 定义在 Room.js的开头
                roomnumber : this.state.roomID ,
                username : this.state.username
            }

            if(this.ws.readyState ==  WebSocket.OPEN){
                this.ws.send(JSON.stringify(data));
                //alert(JSON.stringify(data)); //将send传递的字符串显示出来
            }
        }
        

    }

    /**
     * 功能 ： 准备
     * 触发 ： 普通玩家（非房主）点击“准备”按钮
     * 
     * 变更为准备状态，并向后端发送消息
     */
    ready(){
        if(!this.state.isReady)
        {
        /**
         * 这里要实现用websocket向后端传达准备的功能
         * 
         * 参数：用户名
         * 
         * 
         */

        let data ={
            code : READY ,
            username : this.state.username
        }

        if(this.ws.readyState ==  WebSocket.OPEN){
            this.ws.send(JSON.stringify(data));
            //alert(JSON.stringify(data)); //将send传递的字符串显示出来
        }

        this.setState(
            {
                isReady : true
            }
        )
        }
        
    }

    /**
     * 功能 ： 取消准备
     * 触发 ： 普通玩家（非房主）点击“取消准备”按钮
     * 
     * 变更为非准备状态，并向后端发送消息
     */
    unready(){
        if(this.state.isReady){

        /**
         * 这里要实现用websocket向后端传达取消准备的功能
         * 
         * 参数：用户名
         */

        let data ={
            code : UNREADY ,
            username : this.state.username
        }

        if(this.ws.readyState ==  WebSocket.OPEN){
            this.ws.send(JSON.stringify(data));
            //alert(JSON.stringify(data)); //将send传递的字符串显示出来
        }

        this.setState(
            {
                isReady : false
            }
        )
        }
    }

    /**
     * 功能 ： 更换队伍
     * 触发 ： 玩家点击“更换队伍”按钮
     * 
     * 变更队伍，并向后端发送消息
     */
    changeTeam(){

        if( this.state.team == TEAM_B )
        {
            let data ={
                code : CHANGE_TEAM_TO_A ,
                username : this.state.username
            }
    
            if(this.ws.readyState ==  WebSocket.OPEN){
                this.ws.send(JSON.stringify(data));
                //alert(JSON.stringify(data)); //将send传递的字符串显示出来
            }
         
        }
        else{
            let data ={
                code : CHANGE_TEAM_TO_B ,
                username : this.state.username
            }
    
            if(this.ws.readyState ==  WebSocket.OPEN){
                this.ws.send(JSON.stringify(data));
                //alert(JSON.stringify(data)); //将send传递的字符串显示出来
            }
    
        }
    }


    /**
     * 功能 : 退出房间
     * 触发 ： 点击退出房间按钮
     * 
     * 退出该房间，向后端发送消息
     * 其中要区分房主与普通玩家，用于确认房主身份是否变更
     */
    exitRoom(){

        if(!this.state.host){
            /**
             * 利用websocket向后端发送username表明退出房间
             */

            let data = {
                code : EXIT_BY_USER,
                username : this.state.username
            }
            
            if(this.ws.readyState ==  WebSocket.OPEN){
                this.ws.send(JSON.stringify(data));
                //alert(JSON.stringify(data)); //将send传递的字符串显示出来
            }

            this.gobackMainPage();
        }
        else{
            /**
             * 利用websocket向后端发送username表明退出房间
             * 
             * 且需要更换房主
             * 
             */

            let data ={
                code : EXIT_BY_HOST ,
                username : this.state.username
            }
    
            if(this.ws.readyState ==  WebSocket.OPEN){
                this.ws.send(JSON.stringify(data));
                //alert(JSON.stringify(data)); //将send传递的字符串显示出来
            }
                    
            this.setState(
                {
                    host : false
                }
            )
            this.gobackMainPage();
        }
    }

    render() {
        
        if(this.state.socketState !== WebSocket.OPEN)
            return (
                <View style={base.container}>
                    <AntDesign
                        style={{marginBottom: 20}}
                        name = {'loading1'} 
                        size={30}
                    />
                    <Text style={styles.waiting}>正在建立连接...</Text>
                </View>
            )

        const teamA = [];
        const teamB = [];
    
        const T = (this.state.team == TEAM_B) ? "B->A" : "A->B" ; //更换队伍按钮的文字说明

        this.state.players.forEach((player) =>{
            if(player.playerteam == TEAM_A) {
                if(player.playerstatus) teamA.push([player.username,'Ready']);
                if(!player.playerstatus) teamA.push([player.username,'UnReady']);
            }
            
            if(player.playerteam == TEAM_B) {
                if(player.playerstatus) teamB.push([player.username,'Ready']);
                if(!player.playerstatus) teamB.push([player.username,'UnReady']);
            }
        }
        )

        const ReadyOrNot = !this.state.isReady ? <TouchableOpacity
                                                    onPress = {this.ready} //-----------该属性需要保留！-------------
                                                    style={base.button}>
                                                    <Text
                                                        style={base.btText}>准备</Text>
                                                </TouchableOpacity> 
                                              : <TouchableOpacity
                                                    onPress = {this.unready} //-----------该属性需要保留！-------------
                                                    style={base.button}>
                                                    <Text
                                                        style={base.btText}>取消准备</Text>
                                                </TouchableOpacity> ;

        const StartOrReady = this.state.host ? <TouchableOpacity
                                                    onPress = {this.startGame} //-----------该属性需要保留！-------------
                                                    style={base.button}>
                                                    <Text
                                                        style={base.btText}>开始游戏</Text>
                                                </TouchableOpacity>
                                             :  ReadyOrNot ;

        return (
            <ImageBackground style={base.background}
                source={require('../src/img/bg1.png')}>
                <TouchableOpacity
                    activeOpacity={1.0}  //设置背景被点击时，透明度不变
                    style={{flex:1}}>
                    <View style={header.container}>
                        <View style={header.header}>
                            <View style={header.Head}>
                                <Ionicons
                                    name = {'md-exit'} 
                                    size = {30}
                                    onPress = {this.exitRoom}
                                />
                            </View>
                            <View style={header.End}>
                                <Entypo
                                    name = {'help-with-circle'}
                                    size = {28}
                                    onPress = {this.help}
                                />
                            </View>
                        </View>
                    </View>

                    <View style={base.containerTop}>
                        <Text
                            style={styles.Text}>
                            房间号：{this.state.roomID}
                        </Text>
                        <Text
                            style={styles.Text}>
                            {'  '}房间密码：{this.state.password}
                        </Text>
                    </View>


                    <View style={base.containerTop}>
                        {StartOrReady}
                    
                        <TouchableOpacity
                            onPress = {this.changeTeam} //-----------该属性需要保留！-------------
                            style={base.button}>
                            <Text
                                style={base.btText}>{T}</Text>
                        </TouchableOpacity> 

                        <TouchableOpacity
                            onPress = {this.exitRoom} //-----------该属性需要保留！-------------
                            style={base.button}>
                            <Text
                                style={base.btText}>退出房间</Text>
                        </TouchableOpacity> 
                    </View>

                    <View style={styles.containerRow}>
                        <View style={styles.list}>
                            <SectionList 
                                sections={[
                                    {title: 'A', data: teamA},
                                ]}
                                keyExtractor={(item, index) => item + index}
                                renderItem={this._renderSectionListItem}
                                renderSectionHeader={this._renderSectionHeader}
                                ItemSeparatorComponent={() => <View style={{backgroundColor:'#DEDEDE',height:2}}></View>}  //分割线
                                numColumns={2}
                                columnWrapperStyle={{borderWidth:3, borderColor:'#f4f4f4'}}
                                style={{marginTop:10,marginLeft:5,marginRight:5}}
                            />
                            <SectionList
                                sections={[
                                    {title: 'B', data: teamB},
                                ]}
                                keyExtractor={(item, index) => item + index}
                                renderItem={this._renderSectionListItem}
                                renderSectionHeader={this._renderSectionHeader}
                                ItemSeparatorComponent={() => <View style={{backgroundColor:'#DEDEDE',height:2}}></View>}  //分割线
                                numColumns={2}
                                columnWrapperStyle={{borderWidth:3, borderColor:'black'}}
                                style={{marginTop:10,marginLeft:5,marginRight:5}}
                            />
                        </View>
                    </View>                            
                    
                </TouchableOpacity>
            </ImageBackground>
        );
    }

    _keyExtractor = (item, index) => "index"+item;
 
    _ItemSeparatorComponent= () => (
        <View style={{backgroundColor:'white',height: 1}}/>
    )

    _renderSectionListItem = ({item}) => (
        <FlatList
          data={item}
          numColumns={2}
          renderItem={this._renderItem}
          keyExtractor={this._keyExtractor}
          ItemSeparatorComponent={this._ItemSeparatorComponent}
        />
    )

    _renderItem = ({item}) => {
        var txt = item;
        return (
            <View style={styles.cell}>
                <Text style={styles.cellText}>{txt}</Text>
            </View>
        )
    }

    _renderSectionHeader = ({section}) => {
        var txt = ' Team ' + section.title;
        return (
            <View style={{height: 25, backgroundColor: '#11ffff', justifyContent: 'center'}}>
                <Text style={styles.SectionHeader}>{txt}</Text>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    Text: {
        color: '#000',
        fontWeight: 'bold',
        fontSize: 20,
    },
    containerRow: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'space-around',
        alignItems: 'flex-start',
    },
    RoomTitle:{
        flex: 1,
        justifyContent: 'space-around',
        alignItems: 'center',
    },
    Table: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    waiting: {
        color: '#000',
        fontSize: 18,
        textAlignVertical: 'center', 
        textAlign: 'center', 
    },
    list: {
        height: height-170,
        width: width-40,
        flexDirection: 'row',
        backgroundColor: '#DEDEDE',
    },
    cell: {
        width: (width-55)/4,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#607B8B', 
        //marginHorizontal: 3,
    },
    cellText: {
        height: 30,
        textAlign: 'center', 
        textAlignVertical: 'center', 
        color: '#fff', 
        fontSize: 15,
        marginLeft: 10,
        marginRight: 10,
    },
    SectionHeader: { 
        height: 30,
        textAlign: 'center', 
        textAlignVertical: 'center',
        backgroundColor: '#473C8B',
        color: 'white',
        fontSize: 20
    },
});