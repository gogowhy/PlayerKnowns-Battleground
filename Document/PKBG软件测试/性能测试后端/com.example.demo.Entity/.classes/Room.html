


<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html id="htmlId">
<head>
  <title>Coverage Report :: Room</title>
  <style type="text/css">
    @import "../../.css/coverage.css";
  </style>
</head>

<body>
<div class="header"></div>

<div class="content">
<div class="breadCrumbs">
    [ <a href="../../index.html">all classes</a> ]
    [ <a href="../index.html">com.example.demo.Entity</a> ]
</div>

<h1>Coverage Summary for Class: Room (com.example.demo.Entity)</h1>

<table class="coverageStats">
<tr>
  <th class="name">Class</th>
<th class="coverageStat 
">
  Class, %
</th>
<th class="coverageStat 
">
  Method, %
</th>
<th class="coverageStat 
">
  Line, %
</th>
</tr>
<tr>
  <td class="name">Room</td>
<td class="coverageStat">
  <span class="percent">
    100%
  </span>
  <span class="absValue">
    (1/ 1)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    61.5%
  </span>
  <span class="absValue">
    (8/ 13)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    68.4%
  </span>
  <span class="absValue">
    (13/ 19)
  </span>
</td>
</tr>

</table>

<br/>
<br/>


<div class="sourceCode"><i>1</i>&nbsp;package com.example.demo.Entity;
<i>2</i>&nbsp;
<i>3</i>&nbsp;import javax.persistence.*;
<i>4</i>&nbsp;import java.util.List;
<i>5</i>&nbsp;
<i>6</i>&nbsp;@Entity
<i>7</i>&nbsp;@Table(name=&quot;roominfo&quot;)
<b class="fc"><i>8</i>&nbsp;public class Room {</b>
<i>9</i>&nbsp;
<i>10</i>&nbsp;    @Id
<i>11</i>&nbsp;    @GeneratedValue(strategy = GenerationType.IDENTITY)
<i>12</i>&nbsp;    public  Integer roomid;
<i>13</i>&nbsp;
<i>14</i>&nbsp;    @Column(name = &quot;RoomNumber&quot;)
<i>15</i>&nbsp;    public Integer roomnumber;
<i>16</i>&nbsp;
<i>17</i>&nbsp;    @Column(name = &quot;HostName&quot;)
<i>18</i>&nbsp;    public String hostname;
<i>19</i>&nbsp;
<i>20</i>&nbsp;    @Column(name=&quot;PlayerNumber&quot;)
<i>21</i>&nbsp;    public Integer playernumber;
<i>22</i>&nbsp;
<i>23</i>&nbsp;    @Column(name = &quot;GameStatus&quot;)
<i>24</i>&nbsp;    public Integer gamestatus;
<i>25</i>&nbsp;
<i>26</i>&nbsp;
<i>27</i>&nbsp;    @Column(name = &quot;RoomPassword&quot;)
<i>28</i>&nbsp;    public Integer roompassword;
<i>29</i>&nbsp;
<i>30</i>&nbsp;    public Integer getRoomid() {
<b class="nc"><i>31</i>&nbsp;        return roomid;</b>
<i>32</i>&nbsp;    }
<i>33</i>&nbsp;
<i>34</i>&nbsp;    public void setRoomid(Integer roomid) {
<b class="nc"><i>35</i>&nbsp;        this.roomid = roomid;</b>
<b class="nc"><i>36</i>&nbsp;    }</b>
<i>37</i>&nbsp;
<i>38</i>&nbsp;    public Integer getRoomnumber() {
<b class="fc"><i>39</i>&nbsp;        return roomnumber;</b>
<i>40</i>&nbsp;    }
<i>41</i>&nbsp;
<i>42</i>&nbsp;    public void setRoomnumber(Integer roomnumber) {
<b class="fc"><i>43</i>&nbsp;        this.roomnumber = roomnumber;</b>
<b class="fc"><i>44</i>&nbsp;    }</b>
<i>45</i>&nbsp;
<i>46</i>&nbsp;    public String getHostname() {
<b class="nc"><i>47</i>&nbsp;        return hostname;</b>
<i>48</i>&nbsp;    }
<i>49</i>&nbsp;
<i>50</i>&nbsp;    public void setHostname(String hostname) {
<b class="fc"><i>51</i>&nbsp;        this.hostname = hostname;</b>
<b class="fc"><i>52</i>&nbsp;    }</b>
<i>53</i>&nbsp;
<i>54</i>&nbsp;    public Integer getPlayernumber() {
<b class="nc"><i>55</i>&nbsp;        return playernumber;</b>
<i>56</i>&nbsp;    }
<i>57</i>&nbsp;
<i>58</i>&nbsp;    public void setPlayernumber(Integer playernumber) {
<b class="fc"><i>59</i>&nbsp;        this.playernumber = playernumber;</b>
<b class="fc"><i>60</i>&nbsp;    }</b>
<i>61</i>&nbsp;
<i>62</i>&nbsp;    public Integer getGamestatus() {
<b class="fc"><i>63</i>&nbsp;        return gamestatus;</b>
<i>64</i>&nbsp;    }
<i>65</i>&nbsp;
<i>66</i>&nbsp;    public void setGamestatus(Integer gamestatus) {
<b class="fc"><i>67</i>&nbsp;        this.gamestatus = gamestatus;</b>
<b class="fc"><i>68</i>&nbsp;    }</b>
<i>69</i>&nbsp;
<i>70</i>&nbsp;
<i>71</i>&nbsp;
<i>72</i>&nbsp;    public Integer getRoompassword() {
<b class="nc"><i>73</i>&nbsp;        return roompassword;</b>
<i>74</i>&nbsp;    }
<i>75</i>&nbsp;
<i>76</i>&nbsp;    public void setRoompassword(Integer roompassword) {
<b class="fc"><i>77</i>&nbsp;        this.roompassword = roompassword;</b>
<b class="fc"><i>78</i>&nbsp;    }</b>
<i>79</i>&nbsp;}
</div>
</div>

<div class="footer">
    
    <div style="float:right;">generated on 2019-07-10 14:50</div>
</div>
</body>
</html>
