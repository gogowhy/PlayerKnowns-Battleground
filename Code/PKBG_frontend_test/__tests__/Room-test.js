/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Room from '../Component/Room';

import WS from 'jest-websocket-mock';
//global.WebSocket = WebSocket;
/*
describe('app', () => {
    it('connect websockets response', (done) => {
                const ws = new WebSocket(`ws://localhost:8080`)
            .on('message',(msg) => {
                expect(JSON.parse(msg).code).toEqual(0);
                ws.close();
            })
            .on('close',() => done());
    });
});
*/
const server = new WS('ws://localhost:1234/');

const ws = new WebSocket('ws://localhost:1234/');


test('renders correctly', () => {

ws.send(JSON.stringify({code : 0 , roomID : '114514'}));
ws.send(JSON.stringify({code : 1 , username : 'nmsl'}));
ws.send(JSON.stringify({code : 2 , username : 'nmsl'}));
ws.send(JSON.stringify({code : 3 , username : 'nmsl'}));
ws.send(JSON.stringify({code : 4 , username : 'nmsl'}));
ws.send(JSON.stringify({code : 5 , username : 'nmsl'}));
ws.send(JSON.stringify({code : 6 , username : 'nmsl'}));
server.send(JSON.stringify({code : 0 }));
    const n = {
        state : {
            params : {
                username : 'aaa',
                roomID : '114514',
                password : '4396',
                host : true
            }
        }
    }

  const tree = renderer.create(<Room navigation={n}  />).toJSON();
  expect(tree).toMatchSnapshot();
});
/*
import React from 'react';
import Intro from '../Intro';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<Intro />).toJSON();
  expect(tree).toMatchSnapshot();
});*/
