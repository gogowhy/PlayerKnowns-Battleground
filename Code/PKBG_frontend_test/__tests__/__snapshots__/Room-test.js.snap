// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 30,
        },
        Object {
          "marginBottom": 20,
        },
        Object {
          "fontFamily": "anticon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 18,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    正在建立连接...
  </Text>
</View>
`;
